src/main/scala/akka/persistence/cassandra/journal/CassandraRecovery.scala

```diff
@@ -22,10 +22,10 @@ trait CassandraRecovery extends ActorLogging {
   private[this] val extendedActorSystem = context.system.asInstanceOf[ExtendedActorSystem]
   private implicit val materializer = ActorMaterializer()
 
-  private[this] val queries: CassandraReadJournal =
+  private[this] lazy val queries: CassandraReadJournal =
```

Comment:
This is required due to the initialisation order between the CassandraRecovery trait and CassandraJournal which is a self type of CassandraRecovery trait. 

If the queries attribute was defined as a val and not a lazy val it would be initialised before config (accessed on line 28) is initialised in CassandraJournal. The result would be a NPE. 

To get around the initialisation order between CassandraJournal and CassandraRecovery the queries attributes needs to be defined as either as a def or a lazy val so that the queries attribute can initialise after the config is initialise in CassandraJournal. 

The downside of this approach is that you won't know of failures in creating the CassandraReadJournal until it is first used. There is probably better approaches to tackling this but would require some refactoring.